Riot/Utils/EventFormatter.m

```diff
@@ -408,6 +411,66 @@ - (NSString*)senderAvatarUrlForEvent:(MXEvent*)event withRoomState:(MXRoomState*
     return senderAvatarUrl;
 }
 
+#pragma mark - MXRoomSummaryUpdating
+- (BOOL)session:(MXSession *)session updateRoomSummary:(MXRoomSummary *)summary withStateEvents:(NSArray<MXEvent *> *)stateEvents roomState:(MXRoomState *)roomState
+{
+    BOOL updated = [super session:session updateRoomSummary:summary withStateEvents:stateEvents roomState:roomState];
+    
+    if (BuildSettings.supportFunctionalMembers)
+    {
+        // Customisation for EMS Functional Members
+        if ([self functionalMembersEventFromStateEvents:stateEvents])
+        {
+            MXLogDebug(@"[EventFormatter] The functional members event has been updated.")
+            return YES;
```

Comment:
Can you explain and comment why we need to return YES every time?
Returning YES will trigger more operations that can be expensive like the storage. It would be a pity to do this if there is no actual change.

Reply:
Ah ok. Is my context right here: The docs for `updateRoomSummary:withStateEvents:roomState:` say

> @param stateEvents state events that may change the room summary.

I made the (possibly incorrect) assumption for this to mean that these would only be new/updated state events from a sync response.

Then as we're not storing the service members, the only reliable way to check if they have changed would be to compute the room name. That seemed wasteful to me, given that if there were changes it would get computed twice more in `updateRoomSummary:withServerRoomSummary:roomState:`.

Reply:
ok. Understood. Can you add this comment to the code?